app/src/main/java/sdp/moneyrun/ui/map/MapActivity.java

```diff
@@ -533,6 +535,8 @@ public void checkObjectives(@NonNull Location location) {
         Coin coin = nearestCoin(location, localPlayer.getLocallyAvailableCoins(), THRESHOLD_DISTANCE);
         if (coin != null && !seenCoins.contains(coin)) {
             seenCoins.add(coin);
+            assert riddleDb != null;
```

Comment:
What's the point of the assertion? It will crash regardless, you should handle it with try-catch instead.